web-common/src/features/sources/createModel.ts

```diff
@@ -19,8 +19,8 @@ export async function createModelFromSource(
   folder = removeLeadingSlash(folder);
 
   // Get new model name
-  const modelNames = await getModelNames(queryClient, instanceId);
-  const newModelName = getName(`${sourceName}_model`, modelNames);
+  const allNames = await fetchAllFileNames(queryClient, instanceId);
+  const newModelName = getName(`${sourceName}_model`, allNames);
```

Comment:
**Security Issue**: Changing the method of fetching model names to 'fetchAllFileNames' without proper validation can lead to unauthorized access if the file names include sensitive information or system files. <br> **Fix**: Add validation to ensure that only authorized file names are fetched and processed. Implement filtering based on user permissions and roles. <br> **Code Suggestion**: 
 ```
 -  const allNames = await fetchAllFileNames(queryClient, instanceId);
 -  const newModelName = getName('${sourceName}_model', allNames);
 +  const allNames = await fetchAllFileNames(queryClient, instanceId).filter(name => isValidFileName(name));
 +  const newModelName = getName('${sourceName}_model', allNames.filter(name => userCanAccessFileName(name)));
 ```